File: source/dotnet/subsystem-tests/Models/CalculationType.cs
namespace Energinet.DataHub.Wholesale.Common.Interfaces.Models;

/// <summary>
/// Defines the wholesale calculation type
/// </summary>
public enum CalculationType
{
    /// <summary>
    /// Balance fixing
    /// </summary>
    BalanceFixing = 0,

    /// <summary>
    /// Aggregation.
    /// </summary>
    Aggregation = 1,

    /// <summary>
    /// Wholesale fixing.
    /// </summary>
    WholesaleFixing = 2,

    /// <summary>
    /// First correction settlement.
    /// </summary>
    FirstCorrectionSettlement = 3,

    /// <summary>
    /// Second correction settlement.
    /// </summary>
    SecondCorrectionSettlement = 4,

    /// <summary>
    /// Third correction settlement.
    /// </summary>
    ThirdCorrectionSettlement = 5,
}

File: source/geh_wholesale/src/geh_wholesale/codelists/calculation_type.py
from enum import Enum


class CalculationType(Enum):
    BALANCE_FIXING = "balance_fixing"
    AGGREGATION = "aggregation"
    WHOLESALE_FIXING = "wholesale_fixing"
    FIRST_CORRECTION_SETTLEMENT = "first_correction_settlement"
    SECOND_CORRECTION_SETTLEMENT = "second_correction_settlement"
    THIRD_CORRECTION_SETTLEMENT = "third_correction_settlement"


def is_wholesale_calculation_type(calculation_type: CalculationType) -> bool:
    return (
        calculation_type == CalculationType.WHOLESALE_FIXING
        or calculation_type == CalculationType.FIRST_CORRECTION_SETTLEMENT
        or calculation_type == CalculationType.SECOND_CORRECTION_SETTLEMENT
        or calculation_type == CalculationType.THIRD_CORRECTION_SETTLEMENT
    )
